src/server/rdb_load.cc

```diff
@@ -657,12 +657,37 @@ void RdbLoaderBase::OpaqueObjLoader::CreateHMap(const LoadTrace* ltrace) {
     lp = lpShrinkToFit(lp);
     pv_->InitRobj(OBJ_HASH, kEncodingListPack, lp);
   } else {
-    StringMap* string_map = CompactObj::AllocateMR<StringMap>();
-    string_map->set_time(MemberTimeSeconds(GetCurrentTimeMs()));
+    StringMap* string_map;
+    if (config_.append) {
+      // Note we only use append_ when the map size exceeds kMaxBlobLen,
+      // which is greater than 64 so we'll always use a StringMap set not
+      // listpack.
+      if (pv_->ObjType() != OBJ_HASH) {
```

Comment:
if this happens it's a bug in code, right?
In that case we use DFATAL which is a check-fail in debug and an error in release to signify it should not happen unless our code has bugs

Reply:
sure

Reply:
though in release we still need to break early? (if we just logged an error then the cast will fail?)